### PR TITLE
feature/32 przekierowywanie na logowaniu bazujace na roli, logout

### PR DESCRIPTION
**Pull request by FilipAlbrycht**

### CODE REVIEW

File: projekt/Budowa/src/main/java/org/budowa/flow/manager/ManagerDashboardController.java [hunk 1]
```diff
@@ -3,19 +3,24 @@
...
 public class ManagerDashboardController extends DashboardBaseController {
     private final SessionManager sessionManager = SessionManager.inject();
     private final BuildingsService buildingsService = BuildingsService.inject();
     private final SceneManager sceneManager = SceneManager.inject();
+    private final Router router = Router.inject();
 
-    public void handleLogout(ActionEvent actionEvent) {
+    public void handleLogout(ActionEvent actionEvent) throws IOException {
         this.sessionManager.logout();
+        this.router.goTo(Route.DASHBOARD);
```
Comment (kwiniarski97) [inline projekt/Budowa/src/main/java/org/budowa/flow/manager/ManagerDashboardController.java:23]: shouldn't you got to login scene when you logout?

File: projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java [hunk 2]
```diff
@@ -28,17 +34,19 @@
...
+                textFieldUsername.getText(),
+                textFieldPassword.getText()
+            );
+
+            sessionManager.setUser(user);
+            this.router.goTo(Route.DASHBOARD);
+        } catch (NoResultException noResultException) {
+            displayError("Użytkownik nie insteje lub podano złe dane.");
+        } catch (Exception e) {
+            displayError("Coś poszło nietak");
```
Comment (kwiniarski97) [inline projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java:48]: Typo in "nietak"

File: projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java [hunk 3]
```diff
@@ -28,17 +34,19 @@
...
-            displayError();
-        } else {
-            try {
-                this.router.goTo(Route.DASHBOARD);
-            } catch (IOException exception) {
-                // todo show some alert with an error
-                exception.printStackTrace();
-            }
+        try {
+            User user = usersRepository.findByUsernameAndPassword(
```
Comment (kwiniarski97) [inline projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java:38]: the view should not directly call the database as this introduce to much of a coupling between views and database in order to maintain loose coupling the view should have an abstraction layer (UserService) that would hash the password and then call the repository with the user login and hashed password

File: projekt/Budowa/src/main/java/org/budowa/repositories/UsersRepository.java [hunk 4]
```diff
@@ -20,11 +21,17 @@ public UsersRepository() {
     /**
      * find User by Username
      * @param username
+     * @param password
      * @return User
      */
-    public User findByUsername(String username) {
-        TypedQuery<User> q = em.createQuery("SELECT b FROM User b WHERE b.username = :username", User.class);
+    public User findByUsernameAndPassword(String username, String password) {
+        String encryptedPassword = DigestUtils.md5Hex( password );
```
Comment (kwiniarski97) [inline projekt/Budowa/src/main/java/org/budowa/repositories/UsersRepository.java:28]: Encryption should be done in a service layer. 

The repository layer has as single responsibility - to communicate with database.
All of the business logic should be in a Service in this case either in AuthService or UserService.

Code as it is now breaks the Single Responsibility Principle.

File: projekt/Budowa/src/main/java/org/budowa/router/Router.java [hunk 5]
```diff
@@ -49,7 +49,8 @@ public void goTo(Route route) throws IOException {
                     // todo: throw unauthorized exception or some other custom exception
                     return;
                 }
-                var userRole = this.sessionManager.getUserRole();
+                var userRole = this.sessionManager.getUser().getUserRole();
+                System.out.println("role " + userRole);
```
Comment (kwiniarski97) [inline projekt/Budowa/src/main/java/org/budowa/router/Router.java:53]: Is is needed?

File: projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java [hunk 6]
```diff
@@ -3,14 +3,20 @@
...
 import org.budowa.router.Router;
+import org.budowa.services.SessionManager;
 
+import javax.persistence.NoResultException;
 import java.io.IOException;
 import java.util.Random;
 
 public class LoginScene {
 
+    private final UsersRepository usersRepository = new UsersRepository();
```
Comment (kwiniarski97) [inline projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java:18]: Repository is an injectable so it should be injected using the `inject` method.

File: projekt/Budowa/src/main/java/org/budowa/flow/manager/ManagerDashboardController.java (hunk 1 above)
Comment (FilipAlbrycht) [reply projekt/Budowa/src/main/java/org/budowa/flow/manager/ManagerDashboardController.java:23]: forget about it, done

File: projekt/Budowa/src/main/java/org/budowa/router/Router.java (hunk 5 above)
Comment (FilipAlbrycht) [reply projekt/Budowa/src/main/java/org/budowa/router/Router.java:53]: removed

File: projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java (hunk 6 above)
Comment (FilipAlbrycht) [reply projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java:18]: actually it wasn't but done :)

File: projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java (hunk 3 above)
Comment (FilipAlbrycht) [reply projekt/Budowa/src/main/java/org/budowa/flow/login/LoginScene.java:38]: You're right these layers should be separated. I move this logic to SessionManager to method login

**Pull request merged**